### PR TITLE
forgot-password-screen

### PR DESCRIPTION
**Pull request by at-khoanguyen**

### CODE REVIEW

File: Tutorial screen/tutorial screen/tutorial screen/ViewControllers/ForgotPasswordViewController.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            alert.addAction(UIAlertAction(title: "OK", style: UIAlertActionStyle.default, handler: { (action) in
+                alert.dismiss(animated: true, completion: nil)
+                self.emailTextField.text = nil
+                self.gotoLoginScreen()
+            }))
+            self.present(alert, animated: true, completion: nil)
+        } else {
+            let alert = UIAlertController(title: "Alert", message: "Please check your email!", preferredStyle: UIAlertControllerStyle.alert)
+            alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+            self.present(alert, animated: true, completion: nil)
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/ViewControllers/ForgotPasswordViewController.swift]: Same above

File: Tutorial screen/tutorial screen/tutorial screen/ViewControllers/ForgotPasswordViewController.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+extension ForgotPasswordViewController: UITextFieldDelegate {
+    //Not whitespace
+    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+        return string != " "
+    }
+
+    //Naviga keyboard
+    func textFieldShouldReturn(_ scoreText: UITextField) -> Bool {
+        if scoreText.returnKeyType == .done {
+            self.done()
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/ViewControllers/ForgotPasswordViewController.swift]: Same above

File: Tutorial screen/tutorial screen/tutorial screen/ViewControllers/ForgotPasswordViewController.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        } else {
+            self.view.endEditing(true)
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/ViewControllers/ForgotPasswordViewController.swift]: 1. Omit `self`
2. Why do you call `dismissKeyboard()`?

File: Tutorial screen/tutorial screen/tutorial screen/ViewControllers/Home/HomeViewController.swift [hunk 4]
```diff
@@ -33,7 +33,7 @@ final class HomeViewController: UIViewController {
         let logoutButton = UIButton(frame: CGRect(x: 50, y: 0, width: 100, height: 44))
         logoutButton.addTarget(self, action: Selectors.logOut, for: UIControlEvents.touchUpInside)
         logoutButton.setTitle("Log Out", for: .normal)
-        logoutButton.titleLabel?.textColor = .black
+        logoutButton.setTitleColor(UIColor.blue, for: UIControlState.normal)
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/ViewControllers/Home/HomeViewController.swift]: You can omit `UIColor`, `UIControlState ` prefix here.

File: Tutorial screen/tutorial screen/tutorial screen/ViewControllers/ForgotPasswordViewController.swift [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        doneButton.addTarget(self, action: Selectors.done, for: UIControlEvents.touchUpInside)
+        doneButton.setTitle("Done", for: .normal)
+        doneButton.setTitleColor(UIColor.blue, for: UIControlState.normal)
+
+        let rightBarButton = UIBarButtonItem(customView: doneButton)
+        navigationItem.rightBarButtonItem = rightBarButton
+    }
+
+    // MARK: - Public methos
+    func dismissKeyboard() {
```
Comment (at-quangphu) [inline Tutorial screen/tutorial screen/tutorial screen/ViewControllers/ForgotPasswordViewController.swift]: Please check **protection level** of functions

File: Tutorial screen/tutorial screen/tutorial screen/ViewControllers/ForgotPasswordViewController.swift [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    func gotoLoginScreen() {
+        navigationController?.popToRootViewController(animated: true)
+    }
+
+    func done() {
+        if isValidEmail(testStr: emailTextField.text!) {
+            let alert = UIAlertController(title: "Alert", message: "New password has been sent to your email. Please check it!", preferredStyle: UIAlertControllerStyle.alert)
+            alert.addAction(UIAlertAction(title: "OK", style: UIAlertActionStyle.default, handler: { (action) in
+                alert.dismiss(animated: true, completion: nil)
+                self.emailTextField.text = nil
```
[the remark this answers is not in the document]
Comment (at-khoanguyen) [reply Tutorial screen/tutorial screen/tutorial screen/ViewControllers/ForgotPasswordViewController.swift]: I tried but failed! Please help me :D

File: Tutorial screen/tutorial screen/tutorial screen/ViewControllers/ForgotPasswordViewController.swift (hunk 3 above)
Comment (at-khoanguyen) [reply Tutorial screen/tutorial screen/tutorial screen/ViewControllers/ForgotPasswordViewController.swift]: I can't call dismissKeyboard, beacause it's private. Please help me!

**Pull request merged**